python/pyspark/worker.py

```diff
@@ -644,7 +635,39 @@ def verify_arrow_result(table, assign_cols_by_name, expected_cols_and_types):
             )
 
 
-def wrap_grouped_map_arrow_udf(f, return_type, argspec, runner_conf):
+def verify_arrow_table(table, assign_cols_by_name, expected_cols_and_types):
+    import pyarrow as pa
+
+    if not isinstance(table, pa.Table):
+        raise PySparkTypeError(
+            errorClass="UDF_RETURN_TYPE",
+            messageParameters={
+                "expected": "pyarrow.Table",
+                "actual": type(table).__name__,
+            },
+        )
+
+    verify_arrow_result(table, assign_cols_by_name, expected_cols_and_types)
+
+
+def verify_arrow_batch(batch, assign_cols_by_name, expected_cols_and_types):
+    import pyarrow as pa
+
+    if not isinstance(batch, pa.RecordBatch):
+        raise PySparkTypeError(
+            errorClass="UDF_RETURN_TYPE",
+            messageParameters={
+                "expected": "pyarrow.RecordBatch",
+                "actual": type(batch).__name__,
+            },
+        )
+
+    verify_arrow_result(batch, assign_cols_by_name, expected_cols_and_types)
+
+
+def wrap_grouped_map_arrow_udf(f, return_type, argspec, is_iterator, runner_conf):
```

Comment:
shall we add a dedicated function `def wrap_grouped_map_arrow_iter_udf` for the new eval type?